src/ripple/app/ledger/impl/InboundLedgers.cpp

```diff
@@ -292,10 +298,19 @@ class InboundLedgersImp
                 assert (it.second);
                 acquires.push_back (it);
             }
+            for (auto const& it : mRecentFailures)
+            {
+                if (it.second > 1)
+                    ret[beast::lexicalCastThrow <std::string>(
+                        it.second)][jss::failed] = true;
+                else
+                    ret[to_string (it.first)][jss::failed] = true;
+            }
         }
 
         for (auto const& it : acquires)
         {
+            // getJson is expensive, so call without the lock
```

Comment:
It's trying to explain why the logic is split, not why this part is done after the lock is released. This splitting has cost and without an explanation for why it's needed, someone might easily remove it.